DP Mystery Map/Assets/Scripts/DialogueManager.cs

```diff
@@ -0,0 +1,110 @@
+using Ink.Parsed;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using Ink.Runtime;
+using PlayerInfo;
+using UnityEngine.UI;
+
+public class DialogueManager : MonoBehaviour
+{
+    [SerializeField] private GameObject dialogPanel;
+    
+    [SerializeField] private TextMeshProUGUI dialogText;
+
+    private Ink.Runtime.Story currentDialog;
+
+    public static bool dialogIsPlaying { get; set; }
+
+    private string fullText;
+    private float charRevealDelay = 0.05f;
+
+    public static DialogueManager instance { get; private set; }
+
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else
+        {
+            Destroy(gameObject); // Ensure only one instance exists
+        }
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+
+        if (Player.collectedItems == Item.None)
+        {
+            dialogIsPlaying = true;
+            dialogPanel.SetActive(true);
+        }
+        else
+        {
+            dialogIsPlaying = false;
+            dialogPanel.SetActive(false);
+        }
+    }
+
+    public void EnterDialogueMode(TextAsset inkJSON)
+    {
+        currentDialog = new Ink.Runtime.Story(inkJSON.text);
+        dialogPanel.SetActive(true);
+        dialogIsPlaying = true;
+
+        ContinueStory();
+    }
+
+    private void ExitDialog() 
+    {
+        dialogText.text = "";
+        dialogPanel.SetActive(false);
+        dialogIsPlaying = false;
+    }
+
+    IEnumerator ShowText()
+    {
+        if (dialogText == null)
+        {
+            Debug.LogError("dialogText is not assigned. Make sure to assign it in the Inspector.");
+            yield break;
+        }
+
+        fullText = currentDialog.Continue(); // Retrieve the next portion of text
+
+        for (int i = 0; i < fullText.Length; i++)
+        {
+            dialogText.text = fullText.Substring(0, i + 1);
+            yield return new WaitForSeconds(charRevealDelay);
```

Comment:
This is more of a QoL than a requirement for the review, but would you be able to make the text advance immediately if the player presses one of the keys that continues the text? So if the dialogue is still displaying and the player presses the interact key, make the text display fully immediately